### DISCUSSION TITLE
Markdown is inserting <br> for newlines

### DISCUSSION THREAD

**Discussion by bdnettleton** [requarks/wiki#4070] [Help / Questions] [answered]
I cut and pasted some existing markdown into my new Wiki.js site and the rendered HTML contains <br> for the newlines in the source.  I think that Markdown should take any set of lines together without a blank line and simply put them into a paragraph.  Is there a way to keep Wiki.js from inserting the <br> elements into the HTML?

**Comment by rlukevie** [comment]
I have exactly the same issue. Markdown written in external editors often is written with intentional line breaks in the source, expecting the markdown renderer to put these lines into one. 

Example:

```
Here is the first source line,
and here the second.
```


--> This should be rendered like this:
```
Here is the first source line, and here the second.
```

But the wiki.js markdown renderer returns this:
```
Here is the first source line,
and here the second.
```

**Comment by NGPixel** [comment, marked as answer]
You can disable newlines in the Admin > Rendering > Markdown Core section

**Comment by griznog** [comment]
I went into the markdown core section and disabled this, but my pages are still getting breaks for newlines that in the original files. If it matters, my pages were all imported via the Git connector from a previous foliant/mkdocs site managed with `vim` as the editor and all have paragraphs that are wrapped with linefeeds. I'm really hoping to not have to remove those linefeeds as I do still prefer editing content with `vim` and pushing it to the wiki via the git storage connector. 

Are there any extra steps here I'm missing to disable this? I tried restarting the wiki.js server but that hasn't helped.

**Reply by NGPixel** [reply]
Did you resave the pages afterwards?

**Reply by griznog** [reply]
I've resaved them, moved the linefeeds around and resaved them, deleted all linefeeds and then re-added them via the wiki.js editor and resaved them. Even the preview in the editor still preserves linefeeds as `<br>`.

In the Markdown Core settings I have
* Allow HTML: enabled
* Automatically Convert Links: enabled
* Automatically convert linebreaks: disabled
* Underline emphasis: disabled
* Typographer: disabled
* Quotes style: English

**Reply by NGPixel** [reply]
The editor preview isn't using the rendering settings you've set so don't rely on that. Just a simple save and then look at the final render outside the editor.

I'm not sure what you mean by moving linefeeds around... The setting only prevents non-paragraph line feeds from rendering as `<br>`, unless you end a line with 2 spaces.

Please post a content example and what you're expecting.

**Reply by griznog** [reply]
I guess we'll blame this one on user error. After creating a new page to demonstrate this, the new page works. Now when I go back to old pages, open-modify-save, they also render correctly (for all I've tried so far). Sorry for the noise, I guess I just needed to click harder on `Save` :grin:
